### DISCUSSION TITLE
context size exceed and chat reply failed with error

### DISCUSSION THREAD

**Discussion by meetzuber** [open-webui/open-webui#4505] [General]
# Bug Report
When using one chat session with RAG and ask multiple questions context size keep on increasing and after some time context size exceed above the limit of model max context size and I get error

## Installation Method

using docker compose for installation.

## Environment

- Open WebUI Version: v0.3.11

## Expected Behavior:

there should be an option to choose how many chat history will be passed with user message and older chat history should not be send with user message.

## Actual Behavior:

It is sending all user histry which is exceeding context size and getting error.

## Reproduction Details

**Steps to Reproduce:**
try to use RAG with large documents and generate long answers. and keep on asking question. user models with is having limited context size like 4096 to get this error quickly.

![image](https://github.com/user-attachments/assets/b9af92fa-2031-466d-9343-d06e8655882e)

**Comment by tjbck** [comment]
Hope this helps! https://openwebui.com/f/hub/context_clip_filter

**Reply by meetzuber** [reply]
Thanks for the help
